docs/guide/faq.rst

```diff
@@ -170,6 +258,98 @@ If you're using the GUI, you've probably noticed that buttons next to numeric fi
+
+I just want a current clamp that will deliver a sequence of current pulses at regular intervals. Vector play seems like overkill for this.
+---------------
+
+Right you are. Pick up pulsedistrib.zip, and unzip it into an empty directory. This creates a subdirectory called pulsedistrib, which contains ``Ipulse1.mod``, ``Ipulse2.mod``, ``readme.txt``, and ``test_1_and_2.hoc``. Read ``readme.txt``, compile the mod files, and then use NEURON to load ``test_1_and_2.hoc``, which is a simple demo of these two current pulse generators.
+
+pulsedistrib also contains ``ipulse3.mod``, ``ipulse3rig.ses``, and ``test_3.hoc``, which address the next question in this list.
+
+I want a current clamp that will generate a pulse when I send it an event, or that I can use to produce pulses at precalculated times.
+-----------------------
+
+Then get pulsedistrib.zip, and unzip it. Inside the pulsedistrib subdirectory you'll find ``ipulse3.mod``, ``ipulse3rig.ses``, and ``test_3.hoc`` (and some other files that pertain to the previous question). ``ipulse3.mod`` contains the NMODL code for a current clamp that produces a current pulse when it receives an input event. ``test_3.hoc`` is a simple demo of the Ipulse3 mechanism, and ``ipulse3rig.ses`` is used by ``test_3.hoc`` to create the GUI for a demo of Ipulse3. It uses a NetStim to generate the events that drive the Ipulse3. If you want to drive an Ipulse3 with recorded or precomputed event times, use the VecStim class as described under the topic `Driving a synapse with recorded or precomputed spike events <https://www.neuron.yale.edu/phpBB/viewtopic.php?f=28&t=2117>`_ in the "Hot tips" area of the `NEURON Forum <https://www.neuron.yale.edu/phpBB/>`_.
+
+I have a set of recorded or calculated spike times. How can I use these to drive a postsynaptic mechanism?
+------------------------
+
+Assuming that your synaptic mechanism has a NET_RECEIVE block, so that it is driven by events delivered by a NetCon, I can think of two ways this might be done. Which one to use depends on how many calculated spike times you are dealing with.
+
+If you only have a "few" spikes (up to a few dozen), you could just dump them into the spike queue at the onset of the simulation. Here's how: 
+
+1.
+    Create a Vector and load it with the times at which you want to activate the synaptic mechanism.
+
+2.
+    Then use an :ref:`FIinitializeHandler <finitialize_handler>` that stuffs the spike times into the NetCon's event queue by calling the `NetCon class's event() method <https://nrn.readthedocs.io/en/latest/hoc/modelspec/programmatic/network/netcon.html?highlight=netcon>`_ during initialization.
+
+    For example, if the Vector that holds the event times is syntimes, and the NetCon that drives the synaptic point process is nc, this would work:
+
+For example, if the Vector that holds the event times is syntimes, and the NetCon that drives the synaptic point process is nc, this would work:
+
+.. code::
+    python
```

Comment:
This is HOC... you can get decent formatting by calling it C++... or in an ideal world, translate to Python